pkg/controller/registrysyncer/registrysyncer_test.go

```diff
@@ -0,0 +1,454 @@
+			expected:    fmt.Errorf("failed to get imageStream %s from registry cluster: %w", "ci/applyconfig", fmt.Errorf("imagestreams.image.openshift.io \"applyconfig\" not found")),
+		},
+		{
+			name: "a new tag",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTag.DeepCopy(), applyconfigIS.DeepCopy()}...),
+			appCIClient: bcc(fakeclient.NewFakeClientWithScheme(scheme.Scheme)),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				actualImageStreamImport := &imagev1.ImageStreamImport{}
+				if err := appCIClient.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStreamImport); err != nil {
+					return err
+				}
+				expectedImageStreamImport := &imagev1.ImageStreamImport{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ImageStreamImport",
+						APIVersion: "image.openshift.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace:       "ci",
+						Name:            "applyconfig",
+						ResourceVersion: "1",
+					},
+					Spec: imagev1.ImageStreamImportSpec{
+						Import: true,
+						Images: []imagev1.ImageImportSpec{{
+							From: corev1.ObjectReference{
+								Kind: "DockerImage",
+								Name: "registry.svc.ci.openshift.org/ci/applyconfig@sha256:4ff455dca5145a078c263ebf716eb1ccd1fe6fd41c9f9de6f27a9af9bbb0349d",
+							},
+							To: &corev1.LocalObjectReference{Name: "latest"},
+							ReferencePolicy: imagev1.TagReferencePolicy{
+								Type: imagev1.LocalTagReferencePolicy,
+							},
+						}},
+					},
+					Status: imagev1.ImageStreamImportStatus{
+						Images: []imagev1.ImageImportStatus{
+							{
+								Image: &imagev1.Image{},
+							},
+						},
+					},
+				}
+				if diff := cmp.Diff(expectedImageStreamImport, actualImageStreamImport); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+
+				actualImageStream := &imagev1.ImageStream{}
+				if err := appCIClient.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStream); err != nil {
+					return err
+				}
+				expectedImageStream := &imagev1.ImageStream{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ci",
+						Name:      "applyconfig",
+						Annotations: map[string]string{
+							"release.openshift.io-something": "copied",
+						},
+					},
+				}
+				if diff := cmp.Diff(expectedImageStream, expectedImageStream); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name: "app.ci is newer",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: bcc(fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTag.DeepCopy()}...)),
```

Comment:
ftr, `NewFakeClientWithScheme(scheme.Scheme)` produces the exact same result as `NewFakeClient`. The latter got undeprecated btw. So maybe change it for brevity?

Reply:
Updated.

`undeprecated `?: TIL.